### PR TITLE
[DEV APPROVED] Multiple offices (part deux)

### PR DESCRIPTION
**Pull request by munckymagik**
This set of changes adds validation rules and behaviour, driven by the need to implement add/edit pages in RAD.

### CODE REVIEW

File: spec/support/field_length_validation_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module FieldLengthValidationHelpers
+  # USAGE: expect_length_of(subject, :field_name, 50).to be_valid
+  def expect_length_of(subject, field, length, fill_char: 'x')
+    num_chars_to_fill = (length - subject.send(field).length)
+    new_value = subject.send(field) + (fill_char * num_chars_to_fill)
+    subject.send("#{field}=", new_value)
+
+    expect(subject)
+  end
```
Comment (neoeno) [inline spec/support/field_length_validation_helpers.rb]: neat

**Comment by neoeno** [comment]
:+1:

File: spec/support/field_length_validation_helpers.rb (hunk 1 above)
Comment (munckymagik) [inline spec/support/field_length_validation_helpers.rb]: thanks

File: app/models/office.rb [hunk 2]
```diff
@@ -1,4 +1,52 @@
 class Office < ActiveRecord::Base
   belongs_to :firm
+
+  validates :email_address,
+    presence: true,
+    length: { maximum: 50 },
+    format: { with: /.+@.+\..+/ }
```
Comment (benbarnett) [inline app/models/office.rb]: but what if I ***AM*** a TLD?

Comment (benbarnett) [inline app/models/office.rb]: :trollface: btw

**Comment by benbarnett** [comment]
:+1:

File: app/models/office.rb (hunk 2 above)
Comment (munckymagik) [inline app/models/office.rb]: :) Yes exactly. It stops subversives like me putting in dan@localhost when testing.

**Pull request merged**